javascript-data-model/main.js

```diff
@@ -0,0 +1,48 @@
+/* exported orderHistory */
+
+var orderHistory = [
+  {
+    orderPlaced: 'August 4, 2020',
+    totalPrice: 34.00,
+    shippedTo: 'JS Masher',
+    orderNumber: 11439416898772232,
+    dateDelivered: 'Aug 8, 2020',
+    itemName: 'JavaScript for impatient programmers',
+    itemAuthor: 'Rauschmayer, Dr. Axel',
+    returnWindow: 'Sep 7, 2020',
+    price: 31.55
+  },
+  {
+    orderPlaced: 'July 19, 2020',
+    totalPrice: 44.53,
+    shippedTo: 'JS Masher',
+    orderNumber: 11399842681280257,
+    dateDelivered: 'Jul 20, 2020',
+    itemName: 'The Timeless Way of Building',
+    itemAuthor: 'Alexander, Christopher',
+    returnWindow: 'Aug 19, 2020',
+    price: 41.33
+  },
+  {
+    orderPlaced: 'July 4, 2020',
+    totalPrice: 17.22,
+    shippedTo: 'JS Masher',
+    orderNumber: 11428755579059409,
+    dateDelivered: 'July 7, 2020',
+    itemName: 'Gamecube Controller Adapter. Super Smash Bros Switch Gamecube Adapter for WII U, PC. Support Turbo and Vibration Features. No Driver and No Lag-Gamecube Adapter',
+    itemAuthor: null,
+    returnWindow: 'Aug 5, 2020',
+    price: 15.98
+  },
+  {
+    orderPlaced: 'July 3, 2020',
+    totalPrice: 138.93,
+    shippedTo: 'JS Masher',
+    orderNumber: 11328831772648248,
+    dateDelivered: 'Jul 5, 2020',
+    itemName: ['GameCube Controller - Super Smash Bros. Edition(Nintendo Switch)', 'The Art of Sql'],
```

Comment:
But here, `itemName` holds an array of strings. When the data type of the value differs like this between entries, looping over your data becomes a nightmare. Try to make sure that your data model and the types each value holds are the same between all entries. The data itself will vary, but the structure they are represented in should always be the same.